tests/check/check_test.go

```diff
@@ -33,12 +37,133 @@ func TestCheckMySQL(t *testing.T) {
 	testRunAll(t, "mysql")
 }
 
+func TestCheckLogs(t *testing.T) {
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = "memory"
+
+	observerLogger, logs := observer.New(zap.DebugLevel)
+	serverCtx := &run.ServerContext{
+		Logger: &logger.ZapLogger{
+			Logger: zap.New(observerLogger),
+		},
+	}
+
+	cancel := tests.StartServerWithContext(t, cfg, serverCtx)
+	defer cancel()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr,
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	client := openfgav1.NewOpenFGAServiceClient(conn)
+
+	createStoreResp, err := client.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{
+		Name: "openfga-demo",
+	})
+	require.NoError(t, err)
+
+	storeID := createStoreResp.GetId()
+
+	typedefs := parser.MustParse(`
+	type user
+
+	type document
+	  relations
+	    define viewer: [user] as self
+	`)
+
+	writeModelResp, err := client.WriteAuthorizationModel(context.Background(), &openfgav1.WriteAuthorizationModelRequest{
+		StoreId:         storeID,
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+		TypeDefinitions: typedefs,
+	})
+	require.NoError(t, err)
+
+	authorizationModelID := writeModelResp.GetAuthorizationModelId()
+
+	_, err = client.Write(context.Background(), &openfgav1.WriteRequest{
+		StoreId: storeID,
+		Writes: &openfgav1.TupleKeys{
+			TupleKeys: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:anne"),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	resp, err := client.Check(context.Background(), &openfgav1.CheckRequest{
+		StoreId:  storeID,
+		TupleKey: tuple.NewTupleKey("document:1", "viewer", "user:anne"),
+	})
+	require.NoError(t, err)
+	require.True(t, resp.Allowed)
```

Comment:
This is the "happy path" Check. If we want to add tests for the logs in error cases, we have to change this test quite a bit.

I suggest that you change the test to use a table-driven approach, similar to how we do our unit tests in other parts of the repo ([example](https://github.com/openfga/openfga/blob/ff41e318068dc0fadcb26186a91202bb729eab59/pkg/server/commands/write_test.go#L19-L90))

Here you could do something like this:

```go
         type test struct {
		_name           string
		req             *openfgav1.CheckRequest
		expectedContext map[string]string
	}
	tests := []test{
		{
			_name: "check_success",
			req: &openfgav1.CheckRequest{
				AuthorizationModelId: authorizationModelID,
				StoreId:              storeID,
				TupleKey:             tuple.NewTupleKey("document:1", "viewer", "user:anne"),
			},
			expectedContext: map[string]string{
				"grpc_service": "openfga.v1.OpenFGAService",
				"grpc_method":  "Check",
				"raw_response": `{"allowed":true,"resolution":""}`,
			},
		},
		{
			_name: "check_fail",
			req: &openfgav1.CheckRequest{
				AuthorizationModelId: authorizationModelID,
				StoreId:              "123",
				TupleKey:             tuple.NewTupleKey("document:1", "viewer", "user:anne"),
			},
			expectedContext: map[string]string{
				"grpc_service": "openfga.v1.OpenFGAService",
				"grpc_method":  "Check",
				"raw_response": `{"code":"validation_error","message":"invalid CheckRequest.StoreId: value does not match regex pattern \"^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$\""}`,
			},
		},
	}

	for _, test := range tests {
		t.Run(test._name, func(t *testing.T) {
			_, err = client.Check(context.Background(), test.req)

			// read the last log emitted
			fields := logs.All()[len(logs.All())-1].ContextMap()

			// all the assertions go here
			require.JSONEq(t, test.expectedContext["raw_response"], string(fields["raw_response"].(json.RawMessage)))
		})
	}
```

(That last test is failing because we are not logging anything in case of an error 😱 )

Reply:
@jpadilla Can you create an issue that we are not logging anything in case of 4xxs?